### PR TITLE
[Enhancement] Add background warehouse and use the correct warehouse

### PR DESCRIPTION
**Pull request by yandongxiao**
## Why I'm doing:

1. For background jobs, StarRocks uses a hard-coded approach to handle the Warehouse. For instance, jobs of the Statistics type initiated by TabletStatMgr. A configuration named lake_background_warehouse is provided to support these background operations.
2. Support for MV (Materialized View) jobs to run within a specified Warehouse.
3. Continue to fix certain misuses of the Default Warehouse.
4. Renamed some methods in the StarOSAgent class, the word "Backend" in the method names should be replaced with "Node". "Node" represents backend or compute node.
5. Hide the config of compaction and background warehouse.

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/http/rest/v2/vo/PartitionInfoView.java [hunk 1]
```diff
@@ -177,8 +181,12 @@ public static PartitionView createFrom(PartitionInfo partitionInfo, Partition pa
                     if (lakeTabletOptional.isPresent()) {
                         LakeTablet lakeTablet = lakeTabletOptional.get();
                         try {
+                            WarehouseManager manager = GlobalStateMgr.getCurrentState().getWarehouseMgr();
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/http/rest/v2/vo/PartitionInfoView.java:184]: what's the code path to reach here? how to identify it should be the background warehouse?

File: fe/fe-core/src/main/java/com/starrocks/lake/LakeTable.java [hunk 2]
```diff
@@ -189,8 +191,12 @@ public Status createTabletsForRestore(int tabletNum, MaterializedIndex index, Gl
         List<Long> shardIds = null;
         try {
             // Ignore the parameter replicationNum
+            WarehouseManager warehouseManager = GlobalStateMgr.getCurrentState().getWarehouseMgr();
+            Warehouse warehouse = warehouseManager.getBackgroundWarehouse();
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/lake/LakeTable.java:195]: the same question, how to identify this is a background function call path only?

File: fe/fe-core/src/main/java/com/starrocks/http/rest/v2/vo/PartitionInfoView.java (hunk 1 above)
Comment (yandongxiao) [reply fe/fe-core/src/main/java/com/starrocks/http/rest/v2/vo/PartitionInfoView.java:184]: Good question. When I debug the code, I clear the Nodes in the Default Warehouse and modify the corresponding positions according to the error log. Next, I will organize the code path and explain why it needs to be set to a background warehouse.

Comment (yandongxiao) [reply fe/fe-core/src/main/java/com/starrocks/http/rest/v2/vo/PartitionInfoView.java:184]: In TablePartitionAction, we define an endpoint, /api/v2/catalogs/{%s}/databases/{%s}/tables/{%s}/partition, through which users can obtain detailed information about a Partition.

While traversing a Partition, createFrom converts the Partition's information into a PartitionView. The storagePath field of PartitionView requires a call to getShardInfo to obtain. It use the first tablet to get the related shardInfo.

For such small tasks, placing them in the background warehouse is appropriate.

Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/http/rest/v2/vo/PartitionInfoView.java:184]: maybe not. check about the partition info, if it contains info about the tablet location. If use background warehouse, the replica location will be the nodes in background warehouse, not the nodes in executing warehouse.

**Review by kevincai — DISMISSED** [review]

**Review by kevincai — DISMISSED** [review]

**Review by HangyuanLiu — DISMISSED** [review]

**Review by Seaven — DISMISSED** [review]

**Review by HangyuanLiu — DISMISSED** [review]

**Review by Seaven — DISMISSED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**